tests/test_contrib/test_scatter.py

```diff
@@ -152,12 +160,21 @@ def test_alpha_param(self):
         Test that the user can supply an alpha param on instantiation
         """
         # Instantiate a scatter plot and provide a custom alpha
-        visualizer = ScatterVisualizer(
-            x="temperature", y="light", alpha=0.7
-        )
+        visualizer = ScatterVisualizer(alpha=0.7, features=["a", "b"])
 
+        # Test param gets set correctly
         assert visualizer.alpha == 0.7
 
+        # Mock ax and fit the visualizer
+        visualizer.ax = mock.MagicMock(autospec=True)
+        visualizer.fit(self.X[:, :2], self.y)
```

Comment:
These two lines of code replace the `visualizer.ax` property, which was a `matplotlib.Axes` object, with a `mock.MagicMock` object. The `MagicMock` records all calls made to it and can return mock values or raise exceptions depending on how we want it tested. The `ax.scatter` method is called on `fit`, so we call fit here with some real data.

Reply:
I think your use of mock here is quite clever and it sets the stage for testing if alpha is sent to the scatterplot